File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int i = 0;
        int curr = Integer.MAX_VALUE;
        int c = 0;
        for(int j = 0; j < nums.length; j++) {
            if(nums[j] != curr) {
                curr = nums[j];
                c = 0;
            }
            c += 1;
            if(c < 3)
                nums[i++] = nums[j];
        }
        return i;
    }
}

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        c = 0
        cr = 2 ** 31 - 1
        i = 0
        for j in range(len(nums)):
            if nums[j] != cr:
                cr = nums[j]
                c = 0
            c += 1
            if c < 3:
                nums[i] = nums[j]
                i += 1
        return i 